q2_dada2/_denoise.py

```diff
@@ -137,10 +137,12 @@ def _denoise_helper(biom_fp, track_fp, hashed_feature_ids):
 # new_col_loc is the name of the column the new column will be inserted after
 def _add_percentage_column(df, numerator_col, denominator_col,
                            new_col_loc, new_col_name):
+    new_col_df = df
```

Comment:
This actually doesn't quite work as you would expect:

![image](https://user-images.githubusercontent.com/274668/65527129-05386f00-dea7-11e9-9e26-8f2492144db0.png)

Reply:
Ah ok

Reply:
I'm assuming there is a copy utility

Reply:
BTW, we have _all_ been here, python's shallow vs deep copy is not immediately apparent, until it bites you!

Reply:
`df = df.copy()`

do you really need to preserve the original though? (sorry not looking at the code, just making assumptions based on previous review)

Reply:
> do you really need to preserve the original though?

I would prefer that.